### ISSUE TITLE
python 3.11 support?

### ISSUE THREAD

**Issue by durack1** [conda-forge/cdms2-feedstock#80]
### Comment:

Python 3.11 has just landed in conda-forge see [here](https://anaconda.org/conda-forge/python/files), it would be great if we had py3.11 support for cdms2. I **hope** this doesn't require a lot of work.

Fix https://github.com/PCMDI/cmor/issues/669

**Comment by xylar** [comment]
The migrator is progressing, see: https://conda-forge.org/status/

**Comment by xylar** [comment]
Currently, esmpy is holding up cdms2.

**Comment by xylar** [comment]
https://github.com/conda-forge/esmpy-feedstock/pull/57

**Comment by xylar** [comment]
https://github.com/conda-forge/esmpy-feedstock/pull/57

**Comment by xylar** [comment]
That one may require migrating to a later version of esmpy, which should be possible as of today since I just merged esmf 8.3.1.

**Comment by xylar** [comment]
https://github.com/conda-forge/esmpy-feedstock/pull/58

**Comment by xylar** [comment]
Sorry, if this seems a little stream-of-consciousness.  I am reporting things as I find them out.

**Comment by durack1** [comment]
@xylar no problem - I also saw that [ESMF 8.4.0](https://github.com/esmf-org/esmf/releases/tag/v8.4.0) landed a couple of days back, which solves our Apple M1 problem, so that I presume is another update that is of relevance here (the 8.4.0 update, not the M1 support, which I don't think will ever be achieved for cdms2 - it would likely take a fair number of changes to get that working)

**Comment by mauzey1** [comment]
@xylar Any news on the status of esmpy?  Do we need to wait for https://github.com/conda-forge/esmpy-feedstock/pull/58 to be merged to build CDMS2 for Python 3.11?

**Comment by xylar** [comment]
@mauzey1, yes, I believe we do. And I'm pretty stuck there and haven't been able to find anyone to help.

**Comment by xylar** [comment]
@mauzey1, to be clear, getting ESMPy build for python 3.11 won't help with this issue since cdms2 depend on cdtime, which is not python 3.11 compatible: https://github.com/conda-forge/cdtime-feedstock/pull/52

If you're interested in ESMPy for python 3.11 for some reason other than cdms2, I think it would be better to open an issue on the ESMPy feedstock.
